### ISSUE TITLE
Addon - Default Libraryr - Register default asset shelf categories to use the Default Library categories

### ISSUE THREAD

**Issue by Draise14** [Bforartists/Bforartists#4582]
The new asset shelves for node editors and 3D View now can have default tags, but they must be saved in the startup. We should have our defaults tags and essentials categories listed and stored for user defaults and UX.

**Comment by Draise14** [comment]
@zNightlord do you know how to store these into the startup file? I initially did a startup test with a swap out, but it didn't quite work as intended. Are these potentially stored as a userpref.blend?

EDIT: ok, did a userpref swap out, the categories definately are not stored, seems only in session. Maybe an addon feature needs to be added to the default library addon.

**Comment by zNightlord** [comment]
> @zNightlord do you know how to store these into the startup file? I initially did a startup test with a swap out, but it didn't quite work as intended. Are these potentially stored as a userpref.blend?
> 
> EDIT: ok, did a userpref swap out, the categories definately are not stored, seems only in session. Maybe an addon feature needs to be added to the default library addon.

It should be in something like startup file user preferences, that is how the brush assets are stored unless I miss something 🤔
![image](https://github.com/user-attachments/assets/2665dea2-a538-473a-b5a8-45df4f14d680)
[UI: Option to store enabled asset shelf catalogs in preferences](https://projects.blender.org/blender/blender/pulls/121363)

**Comment by Draise14** [comment]
Ah, now I see how it works. I will have to make the addon define the categories then let the user opt-in to save the cateogries afer first use I guess.. like user recommended categories from the default library.

**Comment by Draise14** [comment]
Yeah, may have to close this, as the default categories are stored in the user preferences, and first use doesn't have this. Unless we make an addon that automatically registers these categories on first use. 

Will convert to an addon task for the Default Library addon.

**Comment by Draise14** [comment]
Found some code they added to add catalogs exposed per default here:

`versioning_userdef.cc`
```
if (!USER_VERSION_ATLEAST(404, 32)) {
    userdef->uiflag |= USER_FILTER_BRUSHES_BY_TOOL; /*BFA Default to on*/

    BKE_preferences_asset_shelf_settings_ensure_catalog_path_enabled(
        userdef, "VIEW3D_AST_brush_gpencil_paint", "Brushes/Grease Pencil Draw/Draw");
    BKE_preferences_asset_shelf_settings_ensure_catalog_path_enabled(
        userdef, "VIEW3D_AST_brush_gpencil_paint", "Brushes/Grease Pencil Draw/Erase");
    BKE_preferences_asset_shelf_settings_ensure_catalog_path_enabled(
        userdef, "VIEW3D_AST_brush_gpencil_paint", "Brushes/Grease Pencil Draw/Utilities");

    BKE_preferences_asset_shelf_settings_ensure_catalog_path_enabled(
        userdef, "VIEW3D_AST_brush_gpencil_sculpt", "Brushes/Grease Pencil Sculpt/Contrast");
    BKE_preferences_asset_shelf_settings_ensure_catalog_path_enabled(
        userdef, "VIEW3D_AST_brush_gpencil_sculpt", "Brushes/Grease Pencil Sculpt/Transform");
    BKE_preferences_asset_shelf_settings_ensure_catalog_path_enabled(
        userdef, "VIEW3D_AST_brush_gpencil_sculpt", "Brushes/Grease Pencil Sculpt/Utilities");
  }
```

**Comment by zNightlord** [comment]
> Found some code they added to add catalogs exposed per default here:
> 
> `versioning_userdef.cc`
> ```
> if (!USER_VERSION_ATLEAST(404, 32)) {
>     userdef->uiflag |= USER_FILTER_BRUSHES_BY_TOOL; /*BFA Default to on*/
> 
>     BKE_preferences_asset_shelf_settings_ensure_catalog_path_enabled(
>         userdef, "VIEW3D_AST_brush_gpencil_paint", "Brushes/Grease Pencil Draw/Draw");
>     BKE_preferences_asset_shelf_settings_ensure_catalog_path_enabled(
>         userdef, "VIEW3D_AST_brush_gpencil_paint", "Brushes/Grease Pencil Draw/Erase");
>     BKE_preferences_asset_shelf_settings_ensure_catalog_path_enabled(
>         userdef, "VIEW3D_AST_brush_gpencil_paint", "Brushes/Grease Pencil Draw/Utilities");
> 
>     BKE_preferences_asset_shelf_settings_ensure_catalog_path_enabled(
>         userdef, "VIEW3D_AST_brush_gpencil_sculpt", "Brushes/Grease Pencil Sculpt/Contrast");
>     BKE_preferences_asset_shelf_settings_ensure_catalog_path_enabled(
>         userdef, "VIEW3D_AST_brush_gpencil_sculpt", "Brushes/Grease Pencil Sculpt/Transform");
>     BKE_preferences_asset_shelf_settings_ensure_catalog_path_enabled(
>         userdef, "VIEW3D_AST_brush_gpencil_sculpt", "Brushes/Grease Pencil Sculpt/Utilities");
>   }
> ```

There is some over at [blendfile.cc](https://github.com/Bforartists/Bforartists/blob/9ef9ce508d0170573d1430f38bbbaeac40eab3d5/source/blender/blenkernel/intern/blendfile.cc#L1505) for default file startup also

**Comment by Draise14** [comment]
Done.
